packages/debug/src/browser/debug-session.tsx

```diff
@@ -456,16 +459,24 @@ export class DebugSession implements CompositeTreeElement {
             || this._threads.values().next().value;
     }
 
-    protected async updateFrames(): Promise<void> {
-        const thread = this._currentThread;
-        if (!thread || thread.frameCount) {
+    protected async updateFrames(stoppedDetails?: StoppedDetails): Promise<void> {
+        if (stoppedDetails && stoppedDetails.allThreadsStopped) {
+            for (const thread of this.threads) {
+                // Mark all thread as needing a frame reload whenever the user opens it.
+                thread.framesDirty = true;
```

Comment:
Do I understand correctly that instead of marking threads as dirty and lazily updating frames, we could also fetch the new frames for all threads at this point?

What you did is both fixing the issue of other threads not being updated, and also optimized it?

Sounds good to me.

Reply:
Yes, you understand correctly. Fetching all frames on stop would be a solution as well.

Reply:
:+1: lazy update is the way to go.